### ISSUE TITLE
When changing `service_plan_id` for any App Service the associated slots don't update to the new Service Plan

### ISSUE THREAD

**Issue by tonedefdev** [hashicorp/terraform-provider-azurerm#17777]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.0.7

### AzureRM Provider Version

3.11.0

### Affected Resource(s)/Data Source(s)

azurerm_windows_function_app_slot, azurerm_linux_function_app_slot, azurerm_windows_web_app_slot, azurerm_linux_web_app_slot

### Terraform Configuration Files

```hcl
resource "azurerm_service_plan" "service_plan" {
  count               = 1
  name                = "linux-sp"
  resource_group_name = var.resource_group_name

  location = local.region[var.location]
  tags     = local.tags

  os_type                      = "Linux"
  sku_name                     = "P1v3"
  worker_count                 = 3
  zone_balancing_enabled       = true
}

resource "azurerm_service_plan" "new_service_plan" {
  count               = 1
  name                = "new-linux-sp"
  resource_group_name = var.resource_group_name

  location = local.region[var.location]
  tags     = local.tags

  os_type                      = "Linux"
  sku_name                     = "P1v3"
  worker_count                 = 3
  zone_balancing_enabled       = true
}

resource "azurerm_linux_web_app" "webapp_linux" {
  count               = 1
  name                = "webapp${count.index}"
  resource_group_name = var.resource_group_name

  location = local.region[var.location]
  tags     = local.tags

  service_plan_id = azurerm_service_plan.service_plan[0].id

  app_settings = merge({
    "WEBSITE_VNET_ROUTE_ALL" = "1"
  }

  identity {
    type = "SystemAssigned"
  }

  site_config {
    always_on = true
  }
}

variable "webapp_slots" {
  type        = map(any)
  default     = {
    test1 = {
      name       = "test1"
      target_app = 0
    }
  }
}

resource "azurerm_linux_web_app_slot" "webapp_linux_slot" {
  for_each = var.webapp_slots
  name     = each.value.name

  app_service_id          = azurerm_linux_web_app.webapp_linux[each.value.target_app].id
  tags                    = local.tags

  identity {
    type = "SystemAssigned"
  }

  site_config {
    always_on = true
  }
}
```

### Debug Output/Panic Output

```shell
Error: creating/updating App Service Slot VNet association between "test2" (App Service "webapp0" / Resource Group "[omitted]") and Virtual Network "vnet": web.AppsClient#CreateOrUpdateSwiftVirtualNetworkConnectionWithCheckSlot: Failure sending request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=<nil> <nil>
with module.webapp.azurerm_app_service_slot_virtual_network_swift_connection.webapp_windows_slot["test2"]
on .terraform/modules/webapp/web_app.tf line 172, in resource "azurerm_app_service_slot_virtual_network_swift_connection" "webapp_windows_slot":
```

### Expected Behaviour

1. An actual error should be returned which the Azure portal shows the following error:
```text
Subnet webapp-vi0 in VNET vnet is already occupied by service /subscriptions/[redacted]/resourceGroups/[redacted]/providers/Microsoft.Web/serverfarms/new-linux-asp.
```
2. The slots should have been updated with the new Service Plan ID instead the slots still show the old Service Plan that it was originally associated with it prior to updating the main Web App with a new Service Plan

### Actual Behaviour

The slots don't update to a new Service Plan when their primary App Service successfully updates to a new Service Plan

### Steps to Reproduce

1. Create a Service Plan
2. Deploy a Web App with slots on the Service Plan
3. `terraform apply`
4. Add a new Service Plan and update the Web App to use the new Service Plan
5. `terraform apply`

**Comment by xiaxyi** [comment]
@tonedefdev Thanks for raising this issue, I tested the behavior in azure portal but seems the service plan of the slot won't be changed along with the web app. Can you double confirm it from your side using the portal?

**Comment by xiaxyi** [comment]
@tonedefdev Good day, just would like to confirm with you to see if you get any chance to check the feature?

**Comment by tonedefdev** [comment]
I just saw the latest provider fixed this issue so this is closed and fixed in PR #20184
